contributors/design-proposals/dynamic-admission-control-configuration.md

```diff
@@ -0,0 +1,218 @@
+    // apiserver removes initilizer from the initializers list of the resource
+    // if the timeout is reached; If "Fail" is set, apiserver returns timeout
+    // error if the timeout is reached.
+    FailurePolicy FailurePolicyType
+
+    // **Optional for alpha implement**
+    // If timeout is reached, the intializer is removed from the resource's
+    // initializer list by the apiserver.
+    // Default to XXX seconds.
+    Timeout *int64
+}
+
+type FailurePolicyType string
+
+const (
+    Ignore FailurePolicyType = "Ignore"
+    Fail FailurePolicyType = "Fail"
+)
+
+type ExternalAdmissionHook struct {
+    // Operations is the list of operations this hook will be invoked on - Create, Update, or *
+    // for all operations. Defaults to '*'.
+    Operations []OperationType
+    // Resources are the resources this hook should be invoked on. '*' is all resources.
+    Resources []Resource
+    // Subresources is list of subresources. If non-empty, this hook should be invoked on 
+    // all combinations of Resources and Subresources. '*' is all subresources.
+    Subresources []string
+
+    // ClientConfig defines how to talk to the hook.
+    ClientConfig AdmissionHookClientConfig
+
+    // FailurePolicy defines how unrecognized errors from the admission endpoint are handled -
+    // allowed values are Ignore, Fail. Default value is Fail
+    FailurePolicy FailurePolicyType
+}
+
+type Resource struct {
+    // Group is the API group the resource belongs to.
+    Group string
+    // Resource is the name of the resource.
+    Resource string
+}
+
+type OperationType string
+
+const (
+    All OperationType = "*"
+    Create OperationType= "Create"
+    Update OperationType= "Update"
+)
+
+// AdmissionHookClientConfig contains the information to make a TLS
+// connection with the webhook
+type AdmissionHookClientConfig struct {
+    // Service is a reference to the service for this webhook. It must communicate
+	// on port 443
+	Service ServiceReference
+	// CABundle is a PEM encoded CA bundle which will be used to validate webhook's server certificate.
+	CABundle []byte
+}
+
+// ServiceReference holds a reference to Service.legacy.k8s.io
+type ServiceReference struct {
+	// Namespace is the namespace of the service
+	Namespace string
+	// Name is the name of the service
+	Name string
+}
+```
+
+## Synchronization of AdmissionControlConfiguration 
+
+If the `initializer admission controller` and the `generic webhook admission
+controller` watch the `AdmissionControlConfiguration` and act upon deltas, their
+cached version of the configuration might be arbitrarily delayed. This makes it
+impossible to predict what initializer/hooks will be applied to newly created
+objects.
+
+We propose two ways to make the behavior of the `initializer admission
+controller` and the `generic webhook admission controller` predictable.
+
+#### 1. Do consistent read of AdmissionControlConfiguration periodically
```

Comment:
@caesarxuchao - I'm completely fine with this option.
I would even go step further, and do:
- let's do consistent read every 1s
- if there wasn't any successful read over last 5s (or 10s) then block everythin

1qps will not be visible and this will give us some tolerance on unsuccessful reads (which may happen from time to time).

WDYT?